src/en/hentairead/src/eu/kanade/tachiyomi/extension/en/hentairead/Hentairead.kt

```diff
@@ -189,14 +189,18 @@ class Hentairead : Madara("HentaiRead", "https://hentairead.com", "en", dateForm
     override fun pageListParse(document: Document): List<Page> {
         launchIO { countViews(document) }
 
-        val pages = document.selectFirst("[id=single-chapter-js-extra]")?.data()
-            ?.substringAfter(":[")
-            ?.substringBefore("],")
-            ?.let { json.decodeFromString<List<PageDto>>("[$it]") }
+        val pageBaseUrl = document.selectFirst("[id=single-chapter-js-extra]")?.data()
+            ?.substringAfter("chapterExtraData = ")
+            ?.substringBefore(";")?.let { json.decodeFromString<ImageBaseUrlDto>(it).baseUrl }
```

Comment:
Use a regex for this. It is more flexible.

Current implementation is equivalent to:
```regex
chapterExtraData = (.*?);
```
but you probably want:
```regex
\bchapterExtraData\s*=\s*(.*);
```